drivers/volume/portworx/portworx.go

```diff
@@ -380,17 +382,24 @@ func (d *portworx) setDriver() error {
 	if err == nil {
 		endpoint = svc.Spec.ClusterIP
 		if err = d.testAndSetEndpoint(endpoint); err == nil {
+			d.refreshEndpoint = false
 			return nil
 		}
+		logrus.Debugf("testAndSetEndpoint failed for %v: %v", endpoint, err)
```

Comment:
Use util.Logf instead of logrus.

Using ginkgo writer allows to use their non-verbose mode. The ways it works is if all tests pass, none of the ginkgo writer output gets displayed. But if a test fails, all of the output gets dumped.

Reply:
ah! my changes are too old! Will fix this.